go/database/mpt/proof.go

```diff
@@ -297,3 +298,121 @@ func createPathFromKeyPrefix(key common.Key, nibbles uint8) Path {
 	copy(res.path[:], key[:])
 	return res
 }
+
+// NewWitnessProof creates a new witness proof.
+func NewWitnessProof() witness.Proof {
+	return witnessProofAdapter{WitnessProof{proofDb{}}}
+}
+
+// WitnessProofToNodes converts the witness proof to a list of strings.
+// Each string is an RLP node of the witness proof.
+func WitnessProofToNodes(proof witness.Proof) []string {
+	p := proof.(witnessProofAdapter).WitnessProof
```

Comment:
The way it is now, it really only works for this one type of proof implementation and crash for all others.

Reply:
right, it was meant to work this way.  If we can create a proof from other scheme, it will produce a `witness.Proof `as well, but if a user uses it to convert as an MPT witness proof, it crashes - it is considered a user mistake. He is suppose to use only a proof instance created from respective scheme.